src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.DeleteCommand;
+import duke.command.DeadlineCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.command.TodoCommand;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parser helper to read and parse raw text input into commands.
+ */
+public class Parser {
+
+    private static String getTaskName(String[] msg) {
+        String input = "";
+        for (int i = 1; i < msg.length; i++) {
+            input += msg[i];
+            if (i < msg.length - 1) input += " ";
+        }
+        return input;
+    }
+
+    /**
+     * Parses raw input and returns corresponding command.
+     *
+     * @param input Raw input string.
+     * @return Command to be executed.
+     * @throws Exception  If there is an error in parsing.
+     */
+    public static Command parse(String input) throws Exception {
```

Comment:
Is it better to throw a DukeException here?

Reply:
While it would be great to throw a DukeException for consistency, some parts of the code can throw other exceptions that I have not yet handled as an DukeException, so I can only throw Exception as a quick fix for now.